internal/elasticsearch/elasticsearch.go

```
@@ -40,6 +46,64 @@ type Database struct {
 	Addr string
 }
 
+// Document represents a document in the elasticsearch database.
+type Document struct {
+	Found   bool            `json:"found"`
+	Id      string          `json:"_id"`
+	Index   string          `json:"_index"`
+	Type    string          `json:"_type"`
+	Version int64           `json:"_version"`
+	Source  json.RawMessage `json:"_source"`
+}
+
+// Alias creates or updates an alias. An alias a is created,
+// or modified if it already exists, to point to i. See
+// http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/indices-aliases.html#indices-aliases
+// for further details.
+func (db *Database) Alias(i, a string) error {
+	indexes, err := db.ListIndexesForAlias(a)
+	if err != nil {
+		return errgo.Notef(err, "error retreiving current aliases")
+	}
+	var actions struct {
+		Actions []action `json:"actions"`
+	}
+	for _, i := range indexes {
+		actions.Actions = append(actions.Actions, action{Remove: &alias{Index: i, Alias: a}})
+	}
+	actions.Actions = append(actions.Actions, action{Add: &alias{Index: i, Alias: a}})
+	if err := db.post(db.url("_aliases"), actions, nil); err != nil {
+		return errgo.Notef(err, "error updating aliases")
+	}
+	return nil
+}
+
+// Create document attempts to create a new document at index/type_/id with the
+// contents in doc. If the document already exists then CreateDocument will return false,
```

I think this might be better to just return an error, but say that if the document already exists, it returns ErrAlreadyExists.

Then the sense of the return value is more obvious and the code using it should look more obvious too.